service/mantle/ups/UpsServices.xml

```diff
@@ -1,6 +1,103 @@
+            <if condition="!postalAddress.shipGatewayAddressId">
+                <if condition="partyId"><then>
+                    <entity-find entity-name="mantle.party.contact.PartyContactMechInfo" list="replacesPcmiList">
+                        <date-filter/><econdition field-name="partyId"/><econdition field-name="replacesContactMechId" from="contactMechId"/>
+                        <select-field field-name="contactMechId"/><order-by field-name="contactMechId"/>
+                    </entity-find>
+                    <if condition="replacesPcmiList">
+                        <set field="contactMechId" from="replacesPcmiList[0].contactMechId"/>
+                        <return/>
+                    </if>
+                </then>
+                    <else-if condition="facilityId">
+                        <entity-find entity-name="mantle.facility.FacilityContactMechInfo" list="replacesFcmiList">
+                            <date-filter/><econdition field-name="facilityId"/><econdition field-name="replacesContactMechId" from="contactMechId"/><select-field field-name="contactMechId"/><order-by field-name="contactMechId"/>
+                        </entity-find>
+                        <if condition="replacesFcmiList">
+                            <set field="contactMechId" from="replacesFcmiList[0].contactMechId"/><return/>
+                        </if>
+                    </else-if>
+                </if>
+            </if>
+
+            <set field="path" value="rest/AV"/>
+            <set field="method" value="POST"/>
+            <entity-find-one entity-name="ups.ShippingGatewayUps" value-field="ups">
+                <field-map field-name="shippingGatewayConfigId"/>
+            </entity-find-one>
+            <set field="accessRequestMap" from="[AccessLicenseNumber:ups.accessLicenceNumber, UserId:ups.userId, Password:ups.password]"/>
+
+            <!-- Preparing address map by calling get#AddressMap service-->
+            <service-call name="mantle.ups.UpsServices.get#AddressMap" out-map="addrOut" in-map="[postalAddress:postalAddress, partyId:partyId]"/>
+
+            <set field="addressMap" from="addrOut.addressMap"/>
+            <set field="addressValidationRequestMap" from="[Request:[TransactionReference:[CustomerContext:'Ups Shipping Address Validation'], RequestAction:'AV'], Address:addressMap]"/>
+            <set field="inMap" from="[AccessRequest:accessRequestMap, AddressValidationRequest:addressValidationRequestMap]"/>
+
+            <!-- Calling generic service for sending request  -->
+            <service-call name="mantle.ups.UpsServices.send#UpsRequest" in-map="[shippingGatewayConfigId:shippingGatewayConfigId, path:path, methodType:method, requestMap:inMap]" out-map="addrValidationResponseOut"/>
+            <set field="responseStatusCode" from="'1'.equals(addrValidationResponseOut.responseMap.AddressValidationResponse.Response?.ResponseStatusCode)"/>
```

Comment:
Please check to use the correct variable name as string comparison is being done here, responseStatusCode is not being collected here.